### DISCUSSION TITLE
loading decorator for widgets

### DISCUSSION THREAD

**Discussion by dfguerrerom** [12rambau/sepal_ui#261] [Ideas]
It would be nice to have a loading decorator that could be used on those widgets that have the `loading` behavior, what do you think @12rambau ?

**Comment by 12rambau** [comment]
I don't fully get what you mean by a "loading decorator" do you think of an extention of the loading_button decorator but for all widgets or is it something else ?

**Comment by 12rambau** [comment]
Ok I've reviewed what you did to AssetSelect I think I get it. Yes that would be a nice feature !

**Comment by dfguerrerom** [comment]
I'm thinking about how to code this feature, but we could use it not only for the `loading` argument, but also for any kind of boolean parameter ( `disabled`... etc) something like.

```Python

@switch('parameter')
def func():
    pass
```

what do you think?
